app/models/project/Tag.scala

```diff
@@ -23,22 +26,47 @@ case class Tag(override val id: Option[Int] = None,
 
   def versionIds: List[Int] = this._versionIds
 
-  def addVersionId(versionId: Int) = {
+  def addVersionId(versionId: Int): Unit = {
     this._versionIds = this._versionIds :+ versionId
     if (isDefined) {
       update(TagVersionIds)
     }
   }
 
   def copyWith(id: Option[Int], theTime: Option[Timestamp]): Tag = this.copy(id = id)
+
+  /**
+    * Used to convert a ghost tag to a normal tag
+    * @author phase
+    */
+  def getFilledTag(service: ModelService)(implicit ex: ExecutionContext): Future[Tag] = {
+    val access = service.access(classOf[Tag])
+    for {
```

Comment:
Just a styling preference, can we get rid of the curly braces for tag and yield here?